blocklylib-core/src/main/java/com/google/blockly/android/ui/AbstractBlockView.java

```diff
@@ -370,7 +373,30 @@ protected boolean isEntireBlockHighlighted() {
      * @return True if the coordinate of the motion event is on the visible, non-transparent part of
      * this view; false otherwise.
      */
-    protected abstract boolean hitTest(MotionEvent event);
+    protected boolean hitTest(MotionEvent event) {
+        int action = event.getAction();
+
+        if (mHasHit && action == MotionEvent.ACTION_MOVE) {
+            // Action started on the block.
+            return true;
+        }
+
+        if (action == MotionEvent.ACTION_CANCEL || action == MotionEvent.ACTION_UP) {
+            // Action has finished
+            boolean wasHit = mHasHit;
+            mHasHit = false;
+            return wasHit;
+        }
+
+        final int eventX = (int) event.getX();
+        final int eventY = (int) event.getY();
+
+        return isInHorizontalRangeOfBlock(eventX) && coordinatesAreOnBlock(eventX, eventY);
+    }
+
+    protected abstract boolean isInHorizontalRangeOfBlock(int x);
```

Comment:
I think this method is just a component of `coordinatesAreOnBlock(..)`. Making it an abstract method imposes assumptions about the spatial layout of the block.